client/pages/sign-up.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <AuthForm v-if="!accountCreated" v-slot="{ invalid }" @submit="register">
+    <AuthAlert :errors="errors">
+      Error creating account, please fix the following errors:
+    </AuthAlert>
+    <InputField
+      v-for="field in fields"
+      :id="field.id"
+      :key="field.index"
+      :field-name="field.name"
+      :field-type="field.type"
+      :field-options="field.options"
+      :rules="field.rules"
+      :inputvalue.sync="field.value"
+    />
+    <AuthSubmit :disabled="invalid">Submit</AuthSubmit>
+    <p class="self-center mt-3">
+      Already have an account?
+      <NuxtLink to="/login" class="text-blue">Click here to log in</NuxtLink>
+    </p>
+  </AuthForm>
+  <AuthForm
+    v-else
+    class="place-items-center gap-5 font-bold text-2xl text-white"
+  >
+    <h1 class="text-3xl">Congratulations {{ name }}!</h1>
+    <p>
+      Your new Elucidate account<wbr />
+      has been created
+    </p>
+    <font-awesome-icon
+      :icon="['fas', 'fa-face-smile-beam']"
+      class="text-[10rem] my-10"
+    />
+    <p>
+      Please proceed to the
+      <NuxtLink to="/quiz" class="text-blue">Home Page</NuxtLink><wbr /> or
+      <NuxtLink to="/login" class="text-blue">Search</NuxtLink>
+      for quizzes
+    </p>
+  </AuthForm>
+</template>
+
+<script>
+let count = 0;
+export default {
+  name: 'SignupPage',
+  auth: false,
+  layout: 'auth',
+  data: () => ({
+    title: 'Sign-Up',
+    accountCreated: false,
+    name: '',
+    errors: {},
+    fields: [
+      {
+        name: 'First Name',
+        type: 'text',
+        index: count++,
+        id: 'first_name',
+        rules: 'required',
+      },
+      {
+        name: 'Last Name',
+        type: 'text',
+        index: count++,
+        id: 'last_name',
+        rules: 'required',
+      },
+      {
+        name: 'Email',
+        type: 'text',
+        index: count++,
+        id: 'email',
+        rules: 'required|email',
+      },
+      {
+        name: 'Password',
+        type: 'password',
+        index: count++,
+        id: 'password',
+        rules: 'required|min:6',
+      },
+      {
+        name: 'Confirm Password',
+        type: 'password',
+        index: count++,
+        id: 'confirm',
+        rules: 'required|password:@password',
+      },
+      {
+        name: 'Grade',
+        type: 'dropdown',
+        options: ['Grade 11', 'Grade 12', 'Other'],
+        index: count++,
+        id: 'grade',
+        rules: 'required',
+      },
+    ],
+  }),
+  head: {
+    title: 'Sign Up',
+  },
+  methods: {
+    async register(e) {
+      const postData = {};
+      this.fields.forEach((field) => {
+        postData[field.id] = e.target.elements[field.name].value;
+      });
+
+      await this.$axios
+        .post('auth/register/', postData)
+        .then(() => {
+          this.title = 'Account Created!';
+          this.name = e.target.elements['First Name'].value;
```

Comment:
This value is set but not actually used, do we need it?

Reply:
See my GitHub comment on line 51.